server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -436,23 +445,26 @@ private CompletionStage<RestResponse> streamEntries(RestRequest request) {
       final MediaType keyMediaType = getMediaType(negotiate, cache, true);
       final MediaType valueMediaType = getMediaType(negotiate, cache, false);
 
-      Cache<?, ?> streamCache = invocationHelper.getRestCacheManager().getCache(cacheName, keyMediaType, valueMediaType, request);
-
-      // Streaming over the cache is blocking
-      return CompletableFuture.supplyAsync(() -> {
-         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
-         CacheStream<? extends Map.Entry<?, ?>> stream = streamCache.entrySet().stream();
-         if (limit > -1) {
-            stream = stream.limit(limit);
-         }
-         responseBuilder.entity(new CacheEntryInputStream(keyMediaType.match(APPLICATION_JSON), valueMediaType.match(APPLICATION_JSON), stream, batch, metadata));
+      AdvancedCache<?, ?> typedCache = invocationHelper.getRestCacheManager().getCache(cacheName, keyMediaType, valueMediaType, request);
+      ComponentRegistry registry = SecurityActions.getComponentRegistry(typedCache);
+      ClusterPublisherManager<?, ?> cpm = registry.getClusterPublisherManager().wired();
+      InvocationContextFactory icf = registry.getComponent(InvocationContextFactory.class);
+      SegmentPublisherSupplier<CacheEntry<?, ?>> sps = cpm.entryPublisher(null, null,
+            icf.createInvocationContext(false, limit), EMPTY_BIT_SET, DeliveryGuarantee.AT_LEAST_ONCE, batch,
```

Comment:
No need for an invocation context in this invocation and we need exactly once when displaying to the console imo.
```suggestion
            null, EMPTY_BIT_SET, DeliveryGuarantee.EXACTLY_ONCE, batch,
```